rules_python/piptool.py

```diff
@@ -551,7 +566,10 @@ def digest(fname):
   return d.hexdigest()
 
 def resolve(args):
-  ordering = build_dep_graph(args)
+  print("Generating %s from %s..." % (args.output, " and ".join(args.input)))
+
+  build_info = ast.literal_eval(args.build_info or '{}')
```

Comment:
Added a comment and changed to json:
```
  # Parse build_info - this is the contents of "requirements_overrides" attribute
  # passed by the user, serialized to json.
  build_info = json.loads(args.build_info or '{}')
```

Reply:
BTW, `literal_eval` is not that bad (compared to `eval`), and I use `literal_eval` anyway for parsing the contents of `requirements.bzl` since that's a starlark dict, not json (trailing commas and comments are not supported by json parser).